src/plugins/GUITestBase/src/GUITestBasePlugin.cpp

```diff
@@ -116,62 +132,49 @@ static int minutes(int minutes) {
-#define REGISTER_TEST_NOT_FOR_MAC(X) REGISTER_TEST_IGNORED_BY(X, HI::GUITest::IgnoredMac, "not for Mac", HI::GUITest::System)
-
-#define REGISTER_TEST_NOT_FOR_LINUX_WITH_TIMEOUT(X, TIMEOUT) REGISTER_TEST_IGNORED_BY_WITH_TIMEOUT(X, TIMEOUT, HI::GUITest::IgnoredLinux, "not for Linux with timeout", HI::GUITest::System)
-#define REGISTER_TEST_NOT_FOR_WINDOWS_WITH_TIMEOUT(X, TIMEOUT) REGISTER_TEST_IGNORED_BY_WITH_TIMEOUT(X, TIMEOUT, HI::GUITest::IgnoredWindows, "not for Windows with timeout", HI::GUITest::System)
-#define REGISTER_TEST_NOT_FOR_MAC_WITH_TIMEOUT(X, TIMEOUT) REGISTER_TEST_IGNORED_BY_WITH_TIMEOUT(X, TIMEOUT, HI::GUITest::IgnoredMac, "not for Mac with timeout", HI::GUITest::System)
-
-#define REGISTER_TEST_LINUX(X, MESSAGE) REGISTER_TEST_IGNORED_BY(X, HI::GUITest::IgnoredWindows | HI::GUITest::IgnoredMac, MESSAGE, HI::GUITest::Bug)
-#define REGISTER_TEST_WINDOWS(X, MESSAGE) REGISTER_TEST_IGNORED_BY(X, HI::GUITest::IgnoredMac | HI::GUITest::IgnoredLinux, MESSAGE, HI::GUITest::Bug)
-#define REGISTER_TEST_MAC(X, MESSAGE) REGISTER_TEST_IGNORED_BY(X, HI::GUITest::IgnoredWindows | HI::GUITest::IgnoredLinux, MESSAGE, HI::GUITest::Bug)
-
-#define REGISTER_TEST_ONLY_LINUX(X) REGISTER_TEST_IGNORED_BY(X, HI::GUITest::IgnoredWindows | HI::GUITest::IgnoredMac, "only for Linux", HI::GUITest::System)
-#define REGISTER_TEST_ONLY_WINDOWS(X) REGISTER_TEST_IGNORED_BY(X, HI::GUITest::IgnoredMac | HI::GUITest::IgnoredLinux, "only for Windows", HI::GUITest::System)
-#define REGISTER_TEST_ONLY_MAC(X) REGISTER_TEST_IGNORED_BY(X, HI::GUITest::IgnoredWindows | HI::GUITest::IgnoredLinux, "only for Mac", HI::GUITest::System)
-
 /** Registers a GUI test with the TIMEOUT and LABELS provided. */
 #define REGISTER_TEST_TL(TestClass, TIMEOUT, LABELS) \
     guiTestBase->registerTest(new TestClass(TIMEOUT, LABELS));
 
-/** Registers a GUI test with the TIMEOUT provided. Adds NIGHTLY_TEST_LABEL label.*/
-#define REGISTER_TEST_T(TestClass, TIMEOUT) REGISTER_TEST_TL(TestClass, TIMEOUT, labels(Nightly));
+/** Registers a GUI test with the TIMEOUT provided as a part of nightly build.*/
+#define REGISTER_TEST_T(TestClass, TIMEOUT) REGISTER_TEST_TL(TestClass, TIMEOUT, nightly());
 
 /** Registers a GUI test with the labels provided.*/
 #define REGISTER_TEST_L(TestClass, LABELS) REGISTER_TEST_TL(TestClass, DEFAULT_GUI_TEST_TIMEOUT, LABELS);
 
-namespace U2 {
+/**  Registers test on all platforms but with a global Ignored label. */
+#define REGISTER_TEST_IGNORED(TestClass, MESSAGE) REGISTER_TEST_EXT(TestClass, nightly(Ignored), MESSAGE)
+/**  Registers test on all platforms but with an IgnoredOnWindows label. */
+#define REGISTER_TEST_IGNORED_WINDOWS(TestClass, MESSAGE) REGISTER_TEST_EXT(TestClass, nightly(IgnoredOnWindows), MESSAGE)
+/**  Registers test on all platforms but with an IgnoredOnMacOS label. */
+#define REGISTER_TEST_IGNORED_MAC(TestClass, MESSAGE) REGISTER_TEST_EXT(TestClass, nightly(IgnoredOnMacOS), MESSAGE)
```

Comment:
REGISTER_TEST_IGNORED_LINUX should be added.

Reply:
This macro will be unused.
The final goal is to remove all ignored macros, so I was happy to remove this one.

Reply:
Ok.